samples/cpp/benchmark_app/remote_blobs_filling.cpp

```diff
@@ -100,45 +81,46 @@ std::map<std::string, std::vector<InferenceEngine::Blob::Ptr>> getRemoteInputBlo
     }
 
-    std::map<std::string, std::vector<InferenceEngine::Blob::Ptr>> remoteBlobs;
-    auto context = exeNetwork.GetContext();
-    auto oclContext = std::dynamic_pointer_cast<InferenceEngine::gpu::ClContext>(context)->get();
-    auto oclInstance = std::make_shared<OpenCL>(oclContext);
+    std::map<std::string, ov::runtime::TensorVector> remoteBlobs;
+    auto context = exeNetwork.get_context();
+    auto& oclContext = static_cast<ov::runtime::gpu::ocl::ClContext&>(context);
+    auto oclInstance = std::make_shared<gpu::OpenCL>(oclContext.get());
 
-    auto setShared = [&](const std::string name, const InferenceEngine::TensorDesc& desc, bool fillRandom = false) {
-        cl_int err;
-        auto inputDims = desc.getDims();
-        auto elementsNum = std::accumulate(begin(inputDims), end(inputDims), 1, std::multiplies<size_t>());
-        auto inputSize = elementsNum * getBytesPerElement(desc.getPrecision());
+    for (auto& inputs_info : app_inputs_info) {
+        for (auto& input : inputs_info) {
+            // Fill random
+            slog::info << "Prepare remote blob for input '" << input.first << "' with random values ("
+                       << std::string((input.second.isImage() ? "image" : "some binary data")) << " is expected)"
+                       << slog::endl;
+
+            auto tensor =
+                oclContext.create_tensor(input.second.precision, input.second.dataShape, clBuffer.back().get());
+            remoteBlobs[input.first].push_back(tensor);
 
-        clBuffer.push_back(cl::Buffer(oclInstance->_context, CL_MEM_READ_WRITE, (cl::size_type)inputSize, NULL, &err));
+            // Creating and filling shared buffers
+            cl_int err;
+            auto elementsNum = std::accumulate(begin(input.second.dataShape),
+                                               end(input.second.dataShape),
+                                               1,
+                                               std::multiplies<size_t>());
+            auto inputSize = elementsNum * input.second.precision.bitwidth() / 8;
+
+            clBuffer.push_back(
+                cl::Buffer(oclInstance->_context, CL_MEM_READ_WRITE, (cl::size_type)inputSize, NULL, &err));
 
-        if (fillRandom) {
             void* mappedPtr = oclInstance->_queue.enqueueMapBuffer(clBuffer.back(),
                                                                    CL_TRUE,
                                                                    CL_MEM_READ_WRITE,
                                                                    0,
                                                                    (cl::size_type)inputSize);
-            fillBuffer(mappedPtr, elementsNum, desc.getPrecision());
+            if (inputFiles.empty()) {
+                // Filling in random data
+                fillBuffer(mappedPtr, elementsNum, input.second.precision);
+            } else {
+                // TODO: add filling with real image data
```

Comment:
Do we have a plan to fix it?

Reply:
original code didn't have that part, there's not that video mem mode supports random only. But generally, yes, we'll add this, but preferably in separate patch.